Update DocsQA Guide (#195)
fixie-ai/ai-jsx: packages/docs/docs/guides/docsqa.md

# DocsQA: Grounding Answers with a Source of Truth

Large Language Models (LLMs) are powerful tools for answering open domain questions, but they're prone to making things up (["hallucination"](<https://en.wikipedia.org/wiki/Hallucination_(artificial_intelligence)>)) when their responses are unrestricted. In addition, publicly available LLMs are trained on public data and consequently won't have knowledge of your internal information (documents, data, documentation, etc.).

One of the common ways to handle both of these issues is to provide the LLM with the content of documents that should be treated as a source of truth. This is called "Document Q&A" or, simply, "DocsQA."

Simple prompting along with the context provided by these additional documents helps avoid hallucination by restricting the domain of responses. At the same time, these additional documents provide new knowledge from internal sources which allows the LLM to give more useful, targeted responses.

Don't worry if this doesn't all make sense yet...keep on reading!

## Preparing a Corpus

> A collection of docs is called a "corpus" (plural "corpora"). Often corpora are backed by a "vector database" that allows for searching over knowledge in a semantically grounded vector space. Once we have this set-up, we can then make queries against the vector DB and make magic happen with an LLM (see [Handling a Query](#handling-a-query) below).

Reviewer: ```suggestion
A collection of docs is called a "corpus" (plural "corpora"). Often corpora are backed by a "vector database" that allows for searching over knowledge in a semantically grounded vector space. Once we have this set-up, we can then make queries against the vector DB to provide docs to an LLM (see [Handling a Query](#handling-a-query) below).
```

nit: I may be oversensitive to this, but the word "magic" in reference to software is a red flag to me.
